src/test/codewhisperer/util/codeParsingUtil.test.ts

```diff
@@ -62,6 +62,186 @@ describe('RegexValidationForJava', () => {
 })
 
 describe('isTestFile', () => {
+    describe('typescript', function () {
+        const language = 'typescript'
+
+        it('isTest by file path should return true', async function () {
+            const filePaths = ['/test/MyClass.ts', '/tst/MyClass.ts', '/tests/MyClass.ts']
+
+            for (const filePath of filePaths) {
+                const result = await isTestFile(filePath, { languageId: language })
+                assert.strictEqual(result, true)
+            }
+        })
+
+        it('isTest by file path should return false', async function () {
+            const filePaths = ['/path/to/MyClass.ts', 'path/to/wrongtesttttfolder/MyClass.ts']
+
+            for (const filePath of filePaths) {
+                const result = await isTestFile(filePath, { languageId: language })
+                assert.strictEqual(result, false)
+            }
+        })
+
+        it('isTest by file name should return true', async function () {
```

Comment:
doesn't need to be a separate test. just combine it with the "matches valid paths" case above